tensorflow_datasets/core/custom_dataset/translate_dataset.py

```diff
@@ -0,0 +1,135 @@
+"""Custom Translate Datasets Template"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+from typing import Dict, List, Tuple, Set
+
+import tensorflow.compat.v2 as tf
+from tensorflow_datasets.core import dataset_builder
+from tensorflow_datasets.core import dataset_info
+from tensorflow_datasets.core import features as features_lib
+from tensorflow_datasets.core import splits as split_lib
+from tensorflow_datasets.core.utils import version
+
+class TranslateDataset(dataset_builder.DatasetBuilder):
+  """Generic image classification dataset created from manual directory.
+
+  The data directory should have the following structure:
+
+  ```
+  path/to/manual_dir/<dataset_name>/
+    split_name/  # Ex: 'train'
+      lang1.txt
+      lang2.txt
+    split_name/  # Ex: 'test'
+      lang1.txt
+      lang2.txt
+      ...
+  ```
+
+  To use it:
+
+  ```
+  builder = tfds.TranslateDataset(root_dir='path/to/manual_dir/')
+  print(builder.info)  # Splits, num examples,... are automatically calculated
+  ds = builder.as_dataset(split='train', shuffle_files=True)
+  ```
+
+  """
+
+  VERSION = version.Version('1.0.0')
+
+  def __init__(self, root_dir: str):
+    root_dir = os.path.expanduser(root_dir)
+    self._languages_examples_dict, languages = _get_language_examples(root_dir)
+    self._languages = sorted(list(languages))
+    super(TranslateDataset, self).__init__()
+    
+    # Reset `_data_dir` as it should not change to DATA_DIR/Version
+    self._data_dir = root_dir
+
+    # Update DatasetInfo splits
+    split_dict = split_lib.SplitDict(self.name)
+    for split_name, examples in self._languages_examples_dict.items():
```

Comment:
As per image_folder.py, I think `_split_examples` would be a better name than `_languages_examples` as it would indicates that dict is mapping `split_name`->`examples`